### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by Alqanar**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+NEXT_PUBLIC_EMAIL_SUBSCRIPTION_FORM_ID=1027
+NEXT_PUBLIC_MARKTO_BASE_URL=//app-ab44.marketo.com
+NEXT_PUBLIC_MARKTO_MUNCHKIN_ID=819-WHT-483
+NEXT_PUBLIC_ENTERPRISE_SIGNUP_FORM_ID=1021
```
Comment (iAdramelk) [inline .env:4]: Let's remove unused keys in the next PRs.

File: .remarkrc.mjs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import remarkVariables from "./.build/server/remark-variables.mjs";
```
Comment (iAdramelk) [inline .remarkrc.mjs:1]: This config is strongly tuned up for docs. We should either remove it or adapt for posts. Also command in the `package.json` points to the docs folder.

File: README.md [hunk 3]
```diff
@@ -1 +1,54 @@
...
+npm install --global yarn
+```
+
+## Installation
+
+Clone the repo and init submodules with the actual docs:
+
+```bash
+git clone git@github.com:gravitational/blog.git
+cd next
```
Comment (iAdramelk) [inline README.md:27]:
```suggestion
cd blog
```

File: README.md [hunk 4]
```diff
@@ -1 +1,54 @@
...
+- `yarn build` - will build static production version.
+- `yarn start` - will display documentation built with `npm run build` at `localhost:3000`.
+
+## Development-related commands
+
+- `yarn test` – runs tests. Used on CI.
+- `yarn lint` – checks JS and TS files for errors and automatically fixes them.
+- `yarn lint-check` – checks JS and TS files for errors, but doesn't fix them. Checked in CI and on commit.
+- `yarn typecheck` – validates TypeScript type-related errors. Used on CI.
+- `yarn build-node` – builds configs and plugins for mdx.
```
Comment (iAdramelk) [inline README.md:50]: We may consider removing `build-node` step for blog and convert files to mjs to simplify build. Its too much boilerplate now. Lets discuss it later.

File: loadr.mjs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// @see https://github.com/lukeed/loadr
+export const loaders = ['tsm'];
```
Comment (iAdramelk) [inline loadr.mjs:2]: This one is used by tests, if we and up not having tests in this repo, we may delete it.

File: next.config.mjs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+const CONTENT_DIRECTORY = resolve("content");
+
+export default withBundleAnalyzer({
+  pageExtensions: ["js", "jsx", "ts", "tsx", "md", "mdx"],
+  outputFileTracing: false,
+  images: {
+    path: "/_next/image",
+    disableStaticImages: true,
+    domains: ["i.ytimg.com"],
+  },
```
Comment (iAdramelk) [inline next.config.mjs:20]: I think we can remove path and domain here.

File: next.config.mjs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  experimental: {
+    // ssr and displayName are configured by default
+    styledComponents: true,
+  },
+  trailingSlash: true,
+  webpack: (config, options) => {
+    config.output.assetModuleFilename = "static/media/[hash][ext]";
```
Comment (iAdramelk) [inline next.config.mjs:27]: I think we can remove this line too.

File: next.config.mjs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        {
+          type: "asset/resource",
+        },
+      ],
+    });
+
+    config.module.rules.push({
+      test: /\.(md|mdx)$/,
+      include: PAGES_DIRECTORY,
+      exclude: CONTENT_DIRECTORY,
```
Comment (iAdramelk) [inline next.config.mjs:51]: We can remove include/exclude here.

File: next.config.mjs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+          options: mdxSiteOptions,
+        },
+      ],
+    });
+
+    config.module.rules.push({
+      test: /\.ya?ml$/,
+      type: "json",
+      use: "yaml-loader",
+    });
```
Comment (iAdramelk) [inline next.config.mjs:65]: Please check if we use yaml files in the blog.

**Review by iAdramelk — APPROVED** [review]

File: .remarkrc.mjs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+if (process.env.WITH_EXTERNAL_LINKS) {
+  configLint.plugins.push([
+    "lint-no-dead-urls",
+    {
+      skipLocalhost: true,
+      skipUrlPatterns: [
+        /teleport\.example\.com/,
+        "http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname",
+        "https://github.com/gravitational/teleport/blob/v{{teleport_version}}/examples/chart/teleport-cluster/templates/clusterrole.yaml",
+        "https://linuxize.com/post/linux-chown-command/",
```
Comment (iAdramelk) [inline .remarkrc.mjs:52]: This is a list of exceptions for docs, blog may not need them.

Comment (Alqanar) [reply .remarkrc.mjs:52]: Okay, thanks, I'll check it out later.

**Pull request merged**